src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/AddRidMetadataAttributeStep.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+
+using Mono.Cecil;
+using Mono.Linker;
+using Mono.Linker.Steps;
+
+#if ILLINK
+using Microsoft.Android.Sdk.ILLink;
+#endif
+
+namespace MonoDroid.Tuner;
+
+public class AddRidMetadataAttributeStep : BaseStep
+{
+	protected override void ProcessAssembly (AssemblyDefinition assembly)
+	{
+		if (!Annotations.HasAction (assembly)) {
+				return;
+		}
+
+		var action = Annotations.GetAction (assembly);
+		if (action == AssemblyAction.Skip || action == AssemblyAction.Delete) {
+			return;
+		}
+
+		string? rid = null;
+#if ILLINK
+		if (!Context.TryGetCustomData ("XARuntimeIdentifier", out rid)) {
+			throw new InvalidOperationException ("Missing XARuntimeIdentifier custom data");
+		}
+#endif
+		if (String.IsNullOrEmpty (rid)) {
+			throw new InvalidOperationException ("RID must have a non-empty value");
+		}
+
+		AssemblyDefinition corlib = GetCorlib ();
+		MethodDefinition assemblyMetadataAttributeCtor = FindAssemblyMetadataAttributeCtor (corlib);
+		TypeDefinition systemString = GetSystemString (corlib);
+
+		var attr = new CustomAttribute (assembly.MainModule.ImportReference (assemblyMetadataAttributeCtor));
+		attr.ConstructorArguments.Add (new CustomAttributeArgument (systemString, "XamarinAndroidAbi")); // key
```

Comment:
Should we avoid naming things with an `XA` or `Xamarin` prefix? Can this just be `RuntimeIdentifier`?

Reply:
`RuntimeIdentifier` is too generic IMO. `XA` or `XamarinAndroid` makes it less likely to clash with other people's metadata

Reply:
Or `AndroidRuntimeIdentifier`? I don't think they want us to introduce new things that say `Xamarin`.

Reply:
Who would know? :)